lib/smooth/application/network/http/regular/MIMEParser.cpp

```diff
@@ -112,6 +150,176 @@ namespace smooth::application::network::http::regular
+                        {
+                            content_callback(id, filename, data.cbegin(), b - 2, first_part, true);  // first_b-2 to
+                                                                                                     // avoid additional
+                                                                                                     // crlf in file end
+                            // std::vector<uint8_t>::iterator e;
+                            // std::advance (e, std::distance<std::vector<uint8_t>::const_iterator>(e, b ) );
+                            // Log::info("end of file",".");
+                            // log_data(data.begin(), e);
+                            // Log::info("the end",".");
+                            status = Status::Headers;
+                            data.erase(data.begin(), b - 2);
+                            first_part = false;
+
+                            if (std::distance<std::vector<uint8_t>::const_iterator>(data.begin(),
+                            find_end_boundary()) < 5)
+                            {
+                                end_of_transmission = true;
+                                data.clear();
+                            }
+                        }
+                    }
+                    else
+                    {
+                        get_more_data = true;
+                    }
+                }
+            }  // while(!end_of_transmission)
+
+            end_of_transmission = false;
+        }
+        else if (mode == Mode::FormURLEncoded)
+        {
+            // URL encoded data can't be parsed in chunks, so wait until all data is received
+            if (data.size() >= expected_content_length)
+            {
+                // Split data on '&' as it comes in. Each part is then expected to contain X=Y, so split on '='.
+                // If a part doesn't contain a '=', put it back in the buffer to be used next time.
+                // The way this works is that split() places any leftovers last in the returned vector
+                // which means that it will be encountered last, thus the loops end at the same time.
+
+                auto parts = split(data, std::vector<uint8_t>{ '&' });
+
+                if (!parts.empty())
+                {
+                    data.clear();
+                }
+
+                URLEncoding encoding{};
+
+                for (const auto& part : parts)
+                {
+                    auto equal_sign = std::find(part.cbegin(), part.cend(), '=');
+
+                    if (equal_sign == part.cend())
+                    {
+                        std::copy(std::make_move_iterator(part.begin()),
+                                  std::make_move_iterator(part.end()),
+                                  std::back_inserter(data));
+                    }
+                    else
+                    {
+                        auto key_value = split(part, std::vector<uint8_t>{ '=' });
```

Comment:
I know this is from my original code, but lets improve it to avoid creating new vector instances each time:
New class-member:
```c++
const std::vector<uint8_t> equal{ '=' };
```

Then:
```c++
auto key_value = split(part, equal);
```

Reply:
o.k. fixed